deps/rabbit/src/rabbit_amqqueue_process.erl

```diff
@@ -1527,13 +1532,21 @@ handle_cast({deliver,
     State1 = State#q{senders = Senders1},
     noreply(maybe_deliver_or_enqueue(Delivery, Delivered, State1));
 
+%% Compat for RabbitMQ 4.2. @todo Remove when 'rabbitmq_4.3.0' FF is required.
 handle_cast({ack, AckTags, ChPid}, State) ->
+    handle_cast({complete, AckTags, ChPid}, State);
+handle_cast({reject, true, AckTags, ChPid}, State) ->
+    handle_cast({requeue, AckTags, ChPid}, State);
+handle_cast({reject, false, AckTags, ChPid}, State) ->
+    handle_cast({discard, AckTags, ChPid}, State);
+
+handle_cast({complete, AckTags, ChPid}, State) ->
     noreply(ack(AckTags, ChPid, State));
 
-handle_cast({reject, true,  AckTags, ChPid}, State) ->
-    noreply(requeue(AckTags, ChPid, State));
+handle_cast({requeue, AckTags, ChPid}, State) ->
+    noreply(requeue(AckTags, true, ChPid, State));
```

Comment:
This boolean looks wrong to me because the `released` outcome must not increment the delivery-count.

Reply:
I pushed a test case which currently fails due to this issue.